### PR TITLE
[SDK-2486]/ InApp class cast exception

### PR DESCRIPTION
**Pull request by william-ct**
Fixes a crash (`ClassCastException`) in header/footer InApp templates.

### CODE REVIEW

**Review by darshanclevertap — APPROVED** [review]
I'm approving this. But @piyush-kukadiya please review this too

File: clevertap-core/src/main/java/com/clevertap/android/sdk/inapp/CTInAppBaseFragment.java [hunk 1]
```diff
@@ -54,7 +56,11 @@ public void onAttach(Context context) {
             config = bundle.getParcelable(Constants.KEY_CONFIG);
             currentOrientation = getResources().getConfiguration().orientation;
             generateListener();
-            didClickForHardPermissionListener = (DidClickForHardPermissionListener) getActivity();
+            /*Initialize the below listener only when in app has InAppNotification activity as their host activity
+            when requesting permission for notification.*/
+            if (context instanceof InAppNotificationActivity) {
```
Comment (piyush-kukadiya) [inline clevertap-core/src/main/java/com/clevertap/android/sdk/inapp/CTInAppBaseFragment.java:61]: @william-ct Please use either `getActivity()` or `context` for if block. Why are you using mixed? I know `context` in `onAttach()` is same as `getActivity()` but why to take risk?

Comment (piyush-kukadiya) [reply clevertap-core/src/main/java/com/clevertap/android/sdk/inapp/CTInAppBaseFragment.java:61]: @william-ct Also you can just say `context  instanceof DidClickForHardPermissionListener`

**Comment by piyush-kukadiya** [comment]
@william-ct You have not updated core sdk version. Please update and run copyTemplates

**Review by darshanclevertap — APPROVED** [review]

**Review by piyush-kukadiya — APPROVED** [review]
👍

**Pull request merged**